### DISCUSSION TITLE
How to add a handler for a custom ImageSource ?

### DISCUSSION THREAD

**Discussion by softlion** [dotnet/maui#8154] [Q&A] [answered]
I created a custom image source (ie: a class derived from `ImageSource`), and a custom `ImageSourceHandler` for each platform (implementing `IImageSourceHandler`).

I used to register it using `ExportImageSourceHandler`.   
So i tried `handlers.AddHandler<CustomImageSource, CustomImageSourceHandler>();` but it complains that CustomImageSourceHandler is not an IIElementHandler.

What is the correct way to register it ?  
I did not find how it is done in Maui for FileImageSource.

**Comment by pictos** [comment, marked as answer]
@softlion you should be able to do it by registering your ImageSourceHandler as a service. [Here's](https://github.com/dotnet/maui/blob/e7812b0f00ffebfe753cc8f35479aa7a4fbf6135/src/Core/src/Hosting/ImageSources/ImageSourcesMauiAppBuilderExtensions.cs#L14-L20) how maui does it

**Reply by softlion** [reply]
Ty I was in the wrong path ! This seems to do it !

**Reply by softlion** [reply]
How maui binds the xaml control `CustomImageSource` (which implements `ICustomImageSource`) to this registration ?  
It detects the first interface implemented by `CustomImageSource` then search it in the services ?  
Or am i missing a piece.

**Reply by pictos** [reply]
`ImageSource` isn't a control. If you're extending from a MAUI control then the MAUI will figure out your imageSourceHandler, I believe

**Reply by softlion** [reply]
Yes sorry for the incorrect word. Anyway I would like to know how the mapping is detected automatically, as there is no link anywhere between the service and the class that implements the interface 

Ex: 
`IFileImageSource` => `FileImageSourceService`
But how it links `IFileImageSource` to the `FileImageSource` class ?

**Reply by pictos** [reply]
@softlion the Image loading is divided into a lot of methods, I don't remember where the services are resolved

**Comment by nanthakumaran-s** [comment]
Hi @softlion, Have you find a way to register the ImageSourceHandler? Stuck at the same issue...
